sklearn/impute/_base.py

```diff
@@ -1053,6 +1053,7 @@ def get_feature_names_out(self, input_features=None):
         feature_names_out : ndarray of str objects
             Transformed feature names.
         """
+        check_is_fitted(self, "n_features_in_")
```

Comment:
The test below is failing because when `addIndicator` is True, the `MissingIndicator` instance doesn't have the attribute `n_features_in`. 
https://github.com/scikit-learn/scikit-learn/blob/2e63d2e41afe419146bae05f6104c08f9f3fdec5/sklearn/impute/tests/test_common.py#L133-L150

Reply:
1. Using `check_is_fitted(self)` will check for the right attributes and the test will pass. 
2. Likewise `check_is_fitted(self, "features_")` will check for the `features_` attribute that is present when `addIndicator` is True.
Any of the above will solve the CI failures.

Reply:
I find the error weird. I am wondering if it is not a bug instead:

- we modify the indicator and not the imputers
- I don't see why the `n_features_in_` should not exist. Adding the indicator will only impact the output.

Reply:
OK the reason is that we pass a pre-computed mask and then we don't validate the input and thus do not create `n_features_in`. It looks like a bug to me.

Reply:
Great, many thanks, @glemaitre . That solves it on my side too.